packages/payment-detection/src/native-token-detector.ts

```diff
@@ -0,0 +1,29 @@
+import { ExtensionTypes, PaymentTypes } from '@requestnetwork/types';
+
+import { ReferenceBasedDetector } from './reference-based-detector';
+import { ReferenceBasedDetectorOptions } from './types';
+
+/**
+ * Handle payment detection for native token payment
+ */
+export abstract class NativeTokenPaymentDetector extends ReferenceBasedDetector<
+  ExtensionTypes.PnReferenceBased.IReferenceBased,
+  PaymentTypes.IETHPaymentEventParameters
+> {
+  public constructor({
```

Comment:
```suggestion
  protected constructor({
```

Reply:
done, see comment for `any-to-native-detector.ts`